src/java/org/apache/cassandra/db/virtual/AccordDebugKeyspace.java

```diff
@@ -0,0 +1,671 @@
+                        return ds;
+                    },
+                    dataSet,
+                    ignore -> false
+                );
+            }
+            return dataSet;
+        }
+    }
+
+    public static final class RejectBeforeTable extends AbstractVirtualTable
+    {
+        private RejectBeforeTable(String keyspace)
+        {
+            super(parse(keyspace,
+                        "Accord per-CommandStore RejectBefore State",
+                        "CREATE TABLE reject_before (\n" +
+                        "  command_store_id int,\n" +
+                           format("range_start %s,\n", ROUTING_KEY_TYPE_STRING) +
+                           format("range_end %s,\n", ROUTING_KEY_TYPE_STRING) +
+                           format("txn_id %s,\n", TXN_ID_TYPE_STRING) +
+                        "  PRIMARY KEY (command_store_id, range_start, range_end)" +
+                        ')'));
+        }
+
+        @Override
+        public DataSet data()
+        {
+            CommandStores stores = ((AccordService) AccordService.instance()).node().commandStores();
+            List<Pair<Integer, RejectBefore>> rangeMaps =
+                getBlockingAndRethrow(stores.map(store -> Pair.create(store.commandStore().id(), store.commandStore().unsafeGetRejectBefore())));
+            rangeMaps.sort(comparing(p -> p.left));
+
+            SimpleDataSet dataSet = new SimpleDataSet(metadata());
+            for (Pair<Integer, RejectBefore> pair : rangeMaps)
+            {
+                int storeId = pair.left;
+                RejectBefore rejectBefore = pair.right;
+
+                if (rejectBefore == null)
+                    continue;
+
+                rejectBefore.foldlWithBounds(
+                    (txnId, ds, start, end) -> ds.row(storeId, decompose(start), decompose(end)).column("txn_id", decompose(txnId)),
+                    dataSet,
+                    ignore -> false
+                );
+            }
+            return dataSet;
+        }
+    }
+
+    public static class TxnBlockedByTable extends AbstractVirtualTable
+    {
+        enum Reason { Self, Txn, Key }
+
+        protected TxnBlockedByTable(String keyspace)
+        {
+            super(parse(keyspace,
+                        "Accord Transactions Blocked By Table" ,
+                        "CREATE TABLE txn_blocked_by (\n" +
+                           format("txn_id %s,\n", TXN_ID_TYPE_STRING) +
+                        "  command_store_id int,\n" +
+                        "  depth int,\n" +
+                           format("blocked_by %s,\n", TXN_ID_TYPE_STRING) +
+                        "  reason text,\n" +
+                        "  save_status text,\n" +
+                           format("execute_at %s,\n", TIMESTAMP_TYPE_STRING) +
+                           format("key %s,\n", ROUTING_KEY_TYPE_STRING) +
```

Comment:
> Fix up existing Accord debug virtual tables

is this the fixup?  Moving off UDT to Tuple?  I am not opposed to this as I think java driver was iffy about handling the UDT.  I am wondering what is the main motivation behind this?

Again, not against, just trying to understand.

Reply:
UDT support for virtual keyspaces was added incorrectly, and adding it correctly is a bit out of scope for this ticket. So I switched from UDT to a tuple - the UDT wasn't really providing much utility either, it's just two elements that are pretty easily human-readable in a tuple without field names.

Reply:
Adding virtual UDTs correctly would involve:
1. Adding a virtual table, `system_virtual_schema.types`, to expose all virtually-registered UDTs per virtual keyspace
2. Modifying the drivers to read virtual UDT definitions from this table

Reply:
as the person who added UDT I am fine switching to tuple, I just wanted to confirm this was the issue

Reply:
> the UDT wasn't really providing much utility either

It was nice in python-dtest but was annoying in java driver... but yeah I am cool with the tuple migration away from UDT